src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+
+    @Nullable
+    private ProductResponseDto getProductResponseDto(ProductRequestDto productRequestDto) {
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+
+        return response.getBody();
+    }
+
+    @NotNull
+    private static ProductRequestDto getProductRequestDto(ResourcesInUserResponseDto resourcesInUser, User user) {
+
+        List<ResourceQuantityRequestDto> listOfResourcesInProduct = getResourceInProductRequestDtos(resourcesInUser);
+
+        ProductRequestDto productRequestDto = new ProductRequestDto();
+        productRequestDto.setName("TestProduct");
+        productRequestDto.setAuthors(List.of("TestAuthors"));
+        productRequestDto.setDescription("Test");
+        productRequestDto.setOwnerId(user.getId());
+        productRequestDto.setSalePrice(50);
+        productRequestDto.setResourcesContent(listOfResourcesInProduct);
+
+        return productRequestDto;
+    }
+
+    @NotNull
+    private static List<ResourceQuantityRequestDto> getResourceInProductRequestDtos(ResourcesInUserResponseDto resourcesInUser) {
+        ResourceQuantityRequestDto resourceQuantityRequestDto = new ResourceQuantityRequestDto();
+        List<ResourceQuantityRequestDto> listOfResourcesInProduct = new ArrayList<>();
+        resourcesInUser.getResourcesAndQuantities().forEach(r ->
+        {
+            resourceQuantityRequestDto.setId(r.getResource().getId());
+            resourceQuantityRequestDto.setQuantity(5);
+            listOfResourcesInProduct.add(resourceQuantityRequestDto);
+        });
+        return listOfResourcesInProduct;
+    }
+
+    @Nullable
+    private ResourcesInUserResponseDto getResourcesInUserResponseDto(ResourceInUserRequestDto resourceInUserRequestDto) {
+        ResponseEntity<ResourcesInUserResponseDto> createResourceInUser =
+                this.testRestTemplate.postForEntity(getBaseResourceAvailabilityUrl(), resourceInUserRequestDto, ResourcesInUserResponseDto.class);
+
+        System.out.println(createResourceInUser.getBody().getOwner());
+        return createResourceInUser.getBody();
+    }
+
+    @NotNull
+    private static ResourceInUserRequestDto getResourceInUserRequestDto(User user, Gemstone gemstone) {
+        ResourceInUserRequestDto resourceInUserRequestDto = new ResourceInUserRequestDto();
+        resourceInUserRequestDto.setUserId(user.getId());
+        resourceInUserRequestDto.setResourceId(gemstone.getId());
+        resourceInUserRequestDto.setQuantity(20);
+        return resourceInUserRequestDto;
+    }
+
+    @Nullable
+    private Gemstone getGemstone() {
+        ResourceRequestDto resourceRequest = ResourceTestHelper.getGemstoneRequestDto();
+        ResponseEntity<Gemstone> createResource =
+                this.testRestTemplate.postForEntity(getBaseResourceUrl(), resourceRequest, Gemstone.class);
+
+        return createResource.getBody();
```

Comment:
This is a bad name for this method. We are not just getting/creating it as a pojo. This is creating a gemstone in the DB. Furthermore, the response from creating a resource is not a Resource, it's ResourceResponseDto. This would fail if the Dto and the entity have different fields.

Reply:
createGemstoneInDatabase?